docs-util/redocly/plugins/decorators/circular-patch.js

```diff
@@ -0,0 +1,82 @@
+module.exports = CircularPatch
+
+/**
+ * Since ref() is triggered upon reaching a $ref OAS line (leaf),
+ * we want to inverse `schemas` instructions in order to optimize iterators.
+ */
+function preparePatches(schemas) {
+  const patches = []
+  const patchesObj = {}
+  for (const schemaToPatch in schemas) {
+    for (const schemaName of schemas[schemaToPatch]) {
+      if (!patchesObj[schemaName]) {
+        patchesObj[schemaName] = {
+          schemaName,
+          schemaPointer: `#/components/schemas/${schemaName}`,
+          schemaToPatchPointers: [],
+        }
+      }
+      const schemaToPatchPointer = `#/components/schemas/${schemaToPatch}`
+      if (
+        !patchesObj[schemaName].schemaToPatchPointers.includes(
+          schemaToPatchPointer
+        )
+      ) {
+        patchesObj[schemaName].schemaToPatchPointers.push(schemaToPatchPointer)
+      }
+    }
+  }
+  for (const key in patchesObj) {
+    patches.push(patchesObj[key])
+  }
+  return patches
+}
+
+function applyPatch(node, schemaName) {
+  delete node["$ref"]
+  node.type = "object"
+  if (!node.description && schemaName) {
+    node.description = `${schemaName} object.`
+  }
+}
+
+function CircularPatch({ schemas = {}, verbose = false }) {
+  const logs = []
+  const patches = preparePatches(schemas)
+
+  return {
+    ref(ref, ctx, resolved) {
+      if (ctx.type.name.toLowerCase() === "schema") {
+        if (resolved?.location?.pointer) {
+          for (const patch of patches) {
+            if (resolved.location.pointer === patch.schemaPointer) {
+              const ctxSchemaPointer = ctx.location.pointer.match(
+                /#\/components\/schemas\/\w+/
+              )[0]
+              if (patch.schemaToPatchPointers.includes(ctxSchemaPointer)) {
+                applyPatch(ref)
+                if (verbose) {
+                  logs.push(
+                    `${ctxSchemaPointer.substring(
+                      "#/components/schemas/".length
```

Comment:
`"#/components/schemas/".length` can be a `const`

Reply:
Will be easier to read as well 💪

Reply:
So true. Done in 34d7148997ba9a1c78dcb7b8fe3b18725c8d1f62